### PR TITLE
Fixes emitter fixture scoping issues

### PR DESCRIPTION
**Pull request by tmckenzie51**
## What was wrong?
At the end of our tests we have hypothesis deprecation warnings that look like - HypothesisDeprecationWarning: tests/core/utilities/test_event_filter_builder.py::test_match_any_bytes_type_properties uses the 'web3' fixture, which is reset between function calls but not between test cases generated by @given(...). You can change it to a module- or session-scoped fixture if it is safe to reuse; if not we recommend using a context manager inside your test function. See https://docs.pytest.org/en/latest/fixture.html#sharing-test-data for details on fixture scope.

Related to Issue #1729

### How was it fixed?
emitter fixtures ('emitter', 'Emitter', 'EMITTER', EMITTER_ABI, 'EMITTER_RUNTIME', 'EMITTER_CODE')  were changed to have  'module' scope for test_contract_data_filters and function scope otherwise.

### Todo:
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/master/newsfragments/README.md)

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

**Comment by a0969818431** [comment]
`defi`

File: tests/core/contracts/test_contract_buildTransaction.py [hunk 1]
```diff
@@ -1,19 +1,43 @@
...
+    local_filter_middleware,
+)
+from web3.providers.eth_tester import (
+    EthereumTesterProvider,
+)
+
+# -*- coding: utf-8 -*-
+
 
 # Ignore warning in pyethereum 1.6 - will go away with the upgrade
```
Comment (kclowes) [inline tests/core/contracts/test_contract_buildTransaction.py:21]: Looks like this is a duplicate, and we don't need it anyway, so let's delete lines 21-26

File: tests/core/filtering/test_contract_data_filters.py [hunk 2]
```diff
@@ -6,6 +6,83 @@
...
+    CONTRACT_EMITTER_RUNTIME,
+)
+from web3.middleware import (
+    local_filter_middleware,
+)
+from web3.providers.eth_tester import (
+    EthereumTesterProvider,
+)
+
+# Ignore warning in pyethereum 1.6 - will go away with the upgrade
```
Comment (kclowes) [inline tests/core/filtering/test_contract_data_filters.py:22]: These two lines can be removed too!

File: tests/core/filtering/test_contract_on_event_filtering.py [hunk 3]
```diff
@@ -4,10 +4,30 @@
...
 
+from web3 import Web3
+from web3.middleware import (
+    local_filter_middleware,
+)
+from web3.providers.eth_tester import (
+    EthereumTesterProvider,
+)
+
 # Ignore warning in pyethereum 1.6 - will go away with the upgrade
```
Comment (kclowes) [inline tests/core/filtering/test_contract_on_event_filtering.py:15]: And these can be removed :)

File: tests/core/filtering/conftest.py [hunk 4]
```diff
@@ -27,7 +27,6 @@ def tester_snapshot(web3):
 
 
 @pytest.fixture(
-    scope='function',
```
Comment (kclowes) [inline tests/core/filtering/conftest.py:30]: Although it doesn't really matter, I think you can put this line back just to cut down on lines changed here.

File: tests/core/filtering/test_contract_on_event_filtering.py [hunk 5]
```diff
@@ -4,10 +4,30 @@
...
 pytestmark = pytest.mark.filterwarnings("ignore:implicit cast from 'char *'")
 
 
+@pytest.fixture(
```
Comment (kclowes) [inline tests/core/filtering/test_contract_on_event_filtering.py:19]: This looks like the same fixture from `tests/core/filtering/conftest.py`. If we remove it, do the tests fail?

File: tests/core/contracts/test_contract_buildTransaction.py [hunk 6]
```diff
@@ -1,19 +1,43 @@
...
 pytestmark = pytest.mark.filterwarnings("ignore:implicit cast from 'char *'")
 
 
+# Ignore warning in pyethereum 1.6 - will go away with the upgrade
+pytestmark = pytest.mark.filterwarnings("ignore:implicit cast from 'char *'")
+
+
+@pytest.fixture(
```
Comment (kclowes) [inline tests/core/contracts/test_contract_buildTransaction.py:29]: I think this is a different web3 than what we want in the contract tests. Do tests fail if you remove this? Or do any warnings come back?

**Review by kclowes — COMMENTED** [review]
Looking good :boom: !

File: tests/core/filtering/test_contract_on_event_filtering.py (hunk 5 above)
Comment (tmckenzie51) [reply tests/core/filtering/test_contract_on_event_filtering.py]: No failure when they're removed!

File: tests/core/contracts/test_contract_buildTransaction.py (hunk 6 above)
Comment (tmckenzie51) [reply tests/core/contracts/test_contract_buildTransaction.py]: No failure, and no new warnings! :)

**Review by kclowes — APPROVED** [review]
Nice!

**Pull request merged**